### PR TITLE
import podcast data

### PR DESCRIPTION
**Pull request by abeglova**
#### Pre-Flight checklist
- [x] Migrations
  - [x] Migration is backwards-compatible with current production code
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested
- [x] Settings
  - [x] New settings are documented and present in `app.json`
  - [x] New settings have reasonable development defaults, if applicable

#### What are the relevant tickets?
https://github.com/mitodl/open-discussions/issues/2736

#### What's this PR do?
This pr ingests podcast data into the course catalog

#### How should this be manually tested?
Set `OPEN_PODCAST_DATA_BRANCH=ab/test-ingest`
Run
`docker-compose run web ./manage.py backpopulate_podcast_data`
Verify that podcasts and podcast episodes are imported and that the first podcast imported has topics and "Custom podcast title" as the title

### CODE REVIEW

File: course_catalog/etl/loaders.py [hunk 1]
```diff
@@ -589,3 +591,87 @@ def load_content_files(course_run, content_files_json):
         else:
             search_task_helpers.delete_run_content_files(course_run.id)
         return content_files
+
+
+def load_podcast_episode(episode_data, podcast):
+    """Load a podcast_episode into the database"""
```
Comment (noisecapella) [inline course_catalog/etl/loaders.py:597]: Can we add arguments and a return value for this docstring?

File: course_catalog/etl/podcast.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+def github_podcast_config_files():
+    """
+    Function that returns a list of pyGithub files with youtube config channel data
+
+    Returns:
+        A list of pyGithub contentFile objects
+    """
+    github_client = github.Github()
+    repo = github_client.get_repo(CONFIG_FILE_REPO)
```
Comment (noisecapella) [inline course_catalog/etl/podcast.py:27]: Can we make `CONFIG_FILE_REPO` and `CONFIG_FILE_FOLDER` into environment variables?

File: course_catalog/etl/podcast.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+
+def github_podcast_config_files():
+    """
+    Function that returns a list of pyGithub files with youtube config channel data
+
+    Returns:
+        A list of pyGithub contentFile objects
+    """
+    github_client = github.Github()
```
Comment (noisecapella) [inline course_catalog/etl/podcast.py:26]: Probably outside the scope of this PR but we may want to use this with github.mit.edu in the future. And, the anonymous client has a lower limit for API calls than an authenticated one. But I'm not sure we will be doing this frequently enough to matter.

File: course_catalog/etl/podcast.py [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+
+    return podcast_configs
+
+
+def extract():
+    """
+    Function for extracting podcast data
+
+    Returns:
+        A generator
```
Comment (noisecapella) [inline course_catalog/etl/podcast.py:96]: Can we specify the types for `(feed, playlist_config)` too?

File: course_catalog/etl/podcast.py [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+    podcast_configs = []
+
+    for file in github_podcast_config_files():
+        try:
+            podcast_config = yaml.safe_load(file.decoded_content)
+
+            errors = validate_podcast_config(podcast_config)
+
+            if errors:
+                log.error(
```
Comment (noisecapella) [inline course_catalog/etl/podcast.py:77]: If there was an error, should we skip adding the config to `podcast_configs`?

File: course_catalog/etl/podcast.py [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+        feed = bs(response.content, "xml")
+        yield (feed, playlist_config)
+
+
+def transform_episode(rss_data, offered_by, topics):
+    """
+    Transform a podcast episode into our normalized data
+
+    Args:
+        episode_rss (tuple): the extracted episode data
```
Comment (noisecapella) [inline course_catalog/etl/podcast.py:119]: Should be `rss_data`, and it looks like a different type than a tuple from the usage in the code below

File: open_discussions/settings.py [hunk 7]
```diff
@@ -576,6 +576,12 @@
         "task": "course_catalog.tasks.get_micromasters_data",
         "schedule": crontab(minute=30, hour=16),  # 12:30pm EST
     },
+    "update-podcasts": {
+        "task": "course_catalog.tasks.get_podcast_data",
+        "schedule": get_int(
+            "YOUTUBE_FETCH_SCHEDULE_SECONDS", 60 * 60 * 2
```
Comment (noisecapella) [inline open_discussions/settings.py:582]: Can we use a different environment variable for this?

**Comment by noisecapella** [comment]
@pdpinch Was this PR closed by mistake?

**Comment by briangrossman** [comment]
Trying to reopen.

File: course_catalog/etl/loaders.py (hunk 1 above)
Comment (noisecapella) [reply course_catalog/etl/loaders.py:597]: I think you missed this comment

Comment (abeglova) [reply course_catalog/etl/loaders.py:597]: Fixing!

**Pull request merged**